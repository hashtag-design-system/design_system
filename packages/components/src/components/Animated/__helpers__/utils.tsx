import { screen } from "@testing-library/react";

export const checkChildStyle = (element: string | HTMLElement, styleValue: { name: string; value: React.ReactText }) => {
  const { name, value } = styleValue;
  if (typeof element === "string") {
    expect(screen.getByTestId(element).style["_values"][`--${name}`]).toBe(String(value));
  } else {
    expect(element.style["_values"][`--${name}`]).toBe(String(value));
  }
};

export const checkStyleCustomProperties = (element: HTMLElement, arr: string[]) => {
  arr
    .map(item => `--${item}`)
    .forEach((style, i) => {
      expect(element.style[i.toString()]).toBe(style);
    });
};
