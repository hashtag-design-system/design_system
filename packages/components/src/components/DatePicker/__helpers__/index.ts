export * from "./DaysTable/DaysTable";
export * from "./MonthContainer/MonthContainer";
export * from "./MonthDays/MonthDays";
export * from "./reducer";
export * from "./types";