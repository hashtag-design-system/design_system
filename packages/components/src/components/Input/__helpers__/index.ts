export * from "./Base/Base";
export * from "./BaseNumber";
export * from "./FieldContainer/FieldContainer";
export * from "./FixGroup/FixGroup";
export * from "./FloatingPlaceholder/FloatingPlaceholder";
export * from "./HelpTextContainer/HelpTextContainer";
export * from "./InputContainer/InputContainer";
export * from "./OuterFieldContainer/OuterFieldContainer";
export * from "./PasswordBoxes/PasswordBoxes";
