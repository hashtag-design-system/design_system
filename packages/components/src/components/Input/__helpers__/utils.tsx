import { render, RenderOptions } from "@testing-library/react";
import React from "react";
import { InputContextProvider } from "../../../utils/contexts/InputContext";
import { generateInputId } from "../../../utils/hooks/useInputId";
import { InputFProps } from "../index";

export const defaultProps: InputFProps = {
  id: generateInputId(),
  placeholder: "Placeholder",
  floatingplaceholder: true,
  state: "default",
};

type InputCustomRenderOptions = {
  providerProps: InputFProps;
  renderOptions?: RenderOptions;
};

export const inputCustomRender = (ui: React.ReactElement, options?: InputCustomRenderOptions) => {
  return render(
    <InputContextProvider value={{ ...defaultProps, ...options?.providerProps }}>{ui}</InputContextProvider>,
    options?.renderOptions
  );
};
