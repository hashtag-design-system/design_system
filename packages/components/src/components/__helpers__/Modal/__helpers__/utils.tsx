import { screen, waitFor } from "@testing-library/react";
import { CONFIG } from "../../../../config";

export const overlayCheckStyle = async (toContain: string): Promise<void> => {
  const modal = screen.getByTestId("modal");

  // Wait due to animation from `opacity: 0` to `opacity: 1`
  await waitFor(() => {
    expect(modal).toBeVisible();
  });

  expect(modal).toHaveAttribute("style");
  if (toContain.includes("rgba")) {
    expect(modal.style.backgroundColor).toContain(toContain);
  } else {
    expect(modal.style["backdropFilter"]).toContain(toContain);
  }
};

export const createPortalElement = () => {
  const newEl = document.createElement("div");
  newEl.setAttribute("id", CONFIG.PORTAL_ID_SELECTOR);
  newEl.setAttribute("data-testid", CONFIG.PORTAL_ID_SELECTOR);
  document.body.prepend(newEl);
  return newEl;
};