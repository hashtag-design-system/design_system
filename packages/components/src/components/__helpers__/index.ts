export { Animated } from "./Animated/Animated";
export type { ComponentProps, InputAutocompleteType, SelectionInputProps } from "./props";
export { Base, SelectionInputStates } from "./SelectionInput/Base";
export type { SelectionInputFProps, SelectionInputState } from "./SelectionInput/Base";
export { LabelContainer } from "./SelectionInput/LabelContainer";
