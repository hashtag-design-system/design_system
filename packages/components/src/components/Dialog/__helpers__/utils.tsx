import { render, RenderOptions } from "@testing-library/react";
import React from "react";
import { DialogContextProvider, DialogContextType } from "../../../utils/contexts";

export const defaultProps: Omit<DialogContextType, "handleDismiss"> = {
  hasBtnGroup: false,
  confirm: false,
  loading: false,
  allowDismissOnLoading: true,
};

type DialogCustomRenderOptions = {
  providerProps: Partial<DialogContextType>;
  renderOptions?: RenderOptions;
};

export const dialogCustomRender = (ui: React.ReactElement, options?: DialogCustomRenderOptions) => {
  return render(
    // @ts-expect-error
    <DialogContextProvider value={{ ...defaultProps, ...options?.providerProps }}>{ui}</DialogContextProvider>,
    options?.renderOptions
  );
};
