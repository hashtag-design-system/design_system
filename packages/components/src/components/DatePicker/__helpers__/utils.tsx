import dayjs from "dayjs";
import React from "react";
import Select from "../../Select";
import DatePicker, { DatePickerFProps, DatePickerOnChangeInfo } from "../index";

type FProps = DatePickerFProps;

export const TEST_DEFAULT_DATE = dayjs("2021-01-26");

export const DefaultDatePicker: React.FC<FProps> = ({ ...props }) => {
  return <DatePicker defaultOpen defaultDates={[TEST_DEFAULT_DATE]} defaultCalendarDate={TEST_DEFAULT_DATE} {...props} />;
};

export const TestDatePickerButton: React.FC<Pick<DatePickerOnChangeInfo, "selectedDate">> = ({ selectedDate }) => (
  <Select.Button>{selectedDate.length >= 1 ? selectedDate[0].format("DD/MM/YYYY") : ""}</Select.Button>
);
