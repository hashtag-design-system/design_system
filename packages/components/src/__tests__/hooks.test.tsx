import { render, screen } from "@testing-library/react";
import { Base } from "../components/Input/__helpers__";
import { inputCustomRender } from "../components/Input/__helpers__/utils";
import { useClassnames, useInputId } from "../utils/hooks";

const TestComponent: React.FunctionComponent<React.ComponentProps<"input"> & { state?: "basic" | "hover" | "pressed" }> = ({
  state,
  ...props
}) => {
  const id = useInputId(props.id);
  const [classNames, rest] = useClassnames("input", props, { stateToRemove: { state, defaultState: "basic" } });

  return (
    <div>
      <input id={id} className={classNames} data-testid="input" {...rest} />
    </div>
  );
};

describe("useClassNames", () => {
  test("default functionality", () => {
    inputCustomRender(<Base />, { providerProps: { className: "custom-className" } });
    const input = screen.getByTestId("input");

    expect(input).toHaveClass("custom-className");
  });
  test("with stateToRemove option", () => {
    inputCustomRender(<Base />, { providerProps: { state: "default" } });
    const input = screen.getByTestId("input");

    expect(input).not.toHaveClass("default");

    inputCustomRender(<Base />, { providerProps: { state: "focus" } });

    expect(input).not.toHaveClass("focus");
  });
  test("with stateToRemove option, with custom defaultState", () => {
    render(<TestComponent state="basic" />);
    const input = screen.getByTestId("input");

    expect(input).not.toHaveClass("basic");
  });
});
describe("useInputId", () => {
  test("default behaviour", () => {
    render(<TestComponent />);
    const input = screen.getByTestId("input");

    expect(input).toHaveAttribute("id");
    expect(input.id).toHaveLength(5);
  });
  test("with predefined id Prop passed", () => {
    const customId = "inputCustomId";
    render(<TestComponent id={customId} />);
    const input = screen.getByTestId("input");

    expect(input).toHaveAttribute("id");
    expect(input.id).toHaveLength(customId.length);
    expect(input.id).toBe(customId);
    expect(input).toMatchSnapshot();
  });
});
describe("useDisabled", () => {
  test('state="disabled"', () => {
    inputCustomRender(<Base />, { providerProps: { state: "disabled" } });
    const input = screen.getByTestId("input");

    expect(input).toHaveClass("disabled");
    expect(input).toBeDisabled();
    expect(input.getAttribute("aria-disabled")).toBeTruthy();
  });
  test('aria-disabled="true"', () => {
    inputCustomRender(<Base />, { providerProps: { "aria-disabled": true } });
    const input = screen.getByTestId("input");

    expect(input).toBeDisabled();
    expect(input.getAttribute("aria-disabled")).toBeTruthy();
  });
  test("disabled={true}", () => {
    inputCustomRender(<Base />, { providerProps: { disabled: true } });
    const input = screen.getByTestId("input");

    expect(input).toBeDisabled();
    expect(input.getAttribute("aria-disabled")).toBeTruthy();
  });
});
