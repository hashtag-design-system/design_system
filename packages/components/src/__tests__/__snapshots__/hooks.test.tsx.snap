// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useInputId with predefined id Prop passed 1`] = `
<input
  class="input"
  data-testid="input"
  id="inputCustomId"
/>
`;
